src/node/node.service.ts

```diff
@@ -0,0 +1,103 @@
+import { HttpService } from '@nestjs/axios';
+import {
+  ForbiddenException,
+  Injectable,
+  InternalServerErrorException,
+  Logger,
+} from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { BigNumber, ethers } from 'ethers';
+import { catchError, firstValueFrom, map } from 'rxjs';
+import getWeek from 'src/utils/date';
+import { ValidatorData } from 'src/utils/types';
+import * as Dpos from './contracts/Dpos.json';
+
+@Injectable()
+export class NodeService {
+  private readonly logger = new Logger(NodeService.name);
+  private ethersProvider: ethers.providers.JsonRpcProvider;
+  private dposContract: ethers.Contract;
+  constructor(
+    private configService: ConfigService,
+    private readonly httpService: HttpService,
+  ) {
+    this.ethersProvider = new ethers.providers.JsonRpcProvider(
+      this.configService.get<string>('taraProvider'),
+    );
+
+    this.dposContract = new ethers.Contract(
+      this.configService.get<string>('dposAddress'),
+      Dpos,
+      this.ethersProvider,
+    );
+  }
+
+  async noActiveValidators(testnet?: boolean) {
+    let explorerRoot;
+    if (testnet) {
+      explorerRoot = this.configService.get<string>('testnetExplorerRoot');
+    } else {
+      explorerRoot = this.configService.get<string>('explorerRoot');
+    }
+    const explorerApi = `${explorerRoot}/api/nodes?week=${getWeek(
+      1,
+    )}&year=${new Date().getUTCFullYear()}`;
```

Comment:
The year/week thing is too complicated. The API already has these default values. Just call it with `/api/nodes?limit=1000`